src/containers/connected-login-modal/connected-login-modal.js

```diff
@@ -1,26 +1,18 @@
 import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
+import * as LoginModalActions from '../../actions/login-modal-actions';
 import loginModalComponent from '../../components/login-modal';
-import Actions from '../../reducers/actions';
 
 const mapStateToProps = state => ({
   isLoginModalOpen: state.login.isLoginModalOpen,
 });
-const mapDispatchToProps = dispatch => ({
-  closeLoginModal: () => {
-    const action = {
-      type: Actions.login.closeLoginModal,
-    };
-
-    dispatch(action);
-  },
-  swapModal: () => {
-    const action = {
-      type: Actions.login.swapModal,
-    };
-
-    dispatch(action);
-  },
-});
+const mapDispatchToProps = dispatch =>
```

Comment:
We have quite a few different files with essentially the same code apart from what they are returning. Could we refactor them.

Such as a lambda to return a lambda that is a dispatch lambda?

Reply:
I'm not sure what you mean? That's why we extracted the actions so they can be used in multiple files.

Reply:
I'm not sure what you mean? That's why we extracted the actions so they can be used in multiple files.

Reply:
for example the only difference between these functions is the name of the function

export const setRequestInformation = data => ({
  type: Actions.requests.setRequestInformation,
  data,
});
export const requestInProgress = data => ({
  type: Actions.requests.requestInProgress,
  data,
});

export const setTypeOfRequest = data => ({
  type: Actions.requests.setTypeOfRequest,
  data,
});

Reply:
const reduxActionCreator = (actionType) => (data) => {type: actionType, data}

Reply:
Ya if that works we can, but in the requests reducere for example, each specific action updates a specific state, so how do we make it so the state is also specified?

Reply:
Its literally just lambdas and json. There is no magic here about redux in this file in particular. 

export const setRequestInformation = reduxActionCreator(Actions.requests.setRequestInformation)

Reply:
Ya we can pair program over it tomorrow unless you want to look at it today

Reply:
sure, we could pair tomorrow.